Automattic/wp-calypso
# Framework: Remove sites-list from devdocs blocks

This PR removes the `sites-list` usage from devdocs, and deletes the no longer used `sites-list` example. Also, it adds `<Site />` and `<AllSites />` examples to the devdocs blocks page, as they were included in the deleted example and we don't want to lose them.

Testing:
* Verify http://calypso.localhost:3000/devdocs/blocks works properly.
* Verify the sites-list example is indeed no longer used anywhere.
* Verify http://calypso.localhost:3000/devdocs/blocks/all-sites works properly.
* Verify http://calypso.localhost:3000/devdocs/blocks/site works properly.
